meta-cli/src/cli/list.rs

```diff
@@ -0,0 +1,175 @@
+
+use super::serialize::SerializeReportExt;
+use super::{Action, ConfigArgs};
+use crate::config::{Config, PathOption};
+use crate::deploy::actors::console::ConsoleActor;
+use crate::deploy::actors::task::serialize::{SerializeAction, SerializeActionGenerator};
+use crate::deploy::actors::task_manager::{StopReason, TaskManagerInit, TaskSource};
+use crate::interlude::*;
+use clap::Parser;
+use common::typegraph::Typegraph;
+use core::fmt::Debug;
+
+use common::graphql::Query;
+use reqwest::header::{HeaderMap, HeaderValue, AUTHORIZATION};
+
+#[derive(Parser, Debug)]
+pub struct List {
+    #[clap(long)]
+    max_parallel_loads: Option<usize>,
+}
+
+#[async_trait]
+impl Action for List {
+    #[tracing::instrument]
+    async fn run(&self, args: ConfigArgs) -> Result<()> {
+        let dir = args.dir()?;
+        let config_path = args.config.clone();
+
+        let config = Config::load_or_find(config_path, &dir)?;
+
+        let config = Arc::new(config);
+
+        let console = ConsoleActor::new(Arc::clone(&config)).start();
+
+        let action_generator = SerializeActionGenerator::new(
+            config.dir().unwrap_or_log().into(),
+            dir.clone().into(),
+            config
+                .prisma_migrations_base_dir(PathOption::Absolute)
+                .into(),
+            true,
+        );
+
+        let task_source = TaskSource::Discovery(dir.into());
+
+        let mut init = TaskManagerInit::<SerializeAction>::new(
+            config.clone(),
+            action_generator,
+            console,
+            task_source,
+        );
+
+        if let Some(max_parallel_tasks) = self.max_parallel_loads {
+            init = init.max_parallel_tasks(max_parallel_tasks);
+        }
+
+        let report = init.run().await;
+
+        match report.stop_reason {
+            StopReason::Error => bail!("failed"),
+            StopReason::Manual | StopReason::ManualForced => {
+                bail!("cancelled")
+            }
+            StopReason::Natural => {}
+            StopReason::Restart => panic!("restart not supported for serialize"),
+        }
+
+        let tgs = report.into_typegraphs()?;
+
+        self.display_typegrahs(tgs).await?;
+        Ok(())
+    }
+}
+
+#[derive(Debug, Deserialize)]
+struct TypegraphObject {
+    name: String,
+    url: String,
+}
+
+impl List {
+    async fn get_more_info(&self) -> Result<Vec<TypegraphObject>> {
+        let client = reqwest::Client::new();
+
+        let mut headers = HeaderMap::new();
+        headers.insert(
+            AUTHORIZATION,
+            HeaderValue::from_str(&format!("Basic {}", "YWRtaW46cGFzc3dvcmQ="))?, // I don't know if it right, may be get from env
+        );
+
+        let query = r#"
+            query {
+                typegraphs {
+                    name
+                    url
+                }
+            }
+        "#;
+
+        let response = client
+            .post("http://localhost:7891/typegate") // this too, may be get from env
```

Comment:
I think you can also get this from the [config](https://github.com/metatypedev/metatype/blob/1d62c0320fb275ce9efabf772de2d604780105d7/examples/metatype.yaml#L3C5-L3C33),

Reply:
```rust
impl Default for NodeConfig {
    fn default() -> Self {
        NodeConfig {
            url: "http://localhost:7890".parse().unwrap(),
            prefix: None,
            username: None,
            password: None,
            env: HashMap::default(),
            secrets: HashMap::default(),
        }
    }
}
```

the url is not same, the graphql is on 7891,

Reply:
I would suggest you try change your local config to point at `localhost:7891` for testing purposes instead, the more generic we are the better it is. Or maybe use something similar to [this](https://github.com/metatypedev/metatype/blob/main/typegate/tests/e2e/templates/templates_test.ts#L50) for quick testing.

Reply:
I think we need to add the same `--target` flag as in the serialize subcommand.